Python/marshal.c

```diff
@@ -1658,7 +1658,14 @@ r_object(RFILE *p)
         Py_ssize_t idx = r_ref_reserve(flag, p);
         PyObject *stop = NULL;
         PyObject *step = NULL;
-        PyObject *start = r_object(p);
+        PyObject *start = NULL;
+
+        if (idx < 0 && flag) {
```

Comment:
```suggestion
        PyObject *start = NULL;
        if (idx < 0 && flag) {
```

Let's use compact code.